### PR TITLE
Add basic Typescript definitions

### PR DESCRIPTION
**Pull request by sammkj**
**What**:
Wrote some basic TS definitions. This only covers `glamorous` function.

**Why**:
Because every good lib deserves type definitions.

**How**:
Styled-components type definitions were referred when writing these definitions.

### CODE REVIEW

**Comment by kentcdodds** [comment]
@sammkj, if I made you a collaborator on this project, would you promise to maintain this? I don't know TypeScript at all and don't plan to maintain this myself. But I'll merge it if you'll watch the repo and keep it up to date :)

**Comment by sammkj** [comment]
@kentcdodds Sure, I will try to do my best to keep the definitions up-to-date =)

File: typings/glamorous.test.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import glamorous from "../";
+
```
Comment (kentcdodds) [inline typings/glamorous.test.tsx]: Is this file run by something? What is this file here for?

Comment (sammkj) [inline typings/glamorous.test.tsx]: This refers to Glamorous package. Typescript will look for package.json in that directory.

Comment (kentcdodds) [inline typings/glamorous.test.tsx]: Hmmm.... So is this file not run at all? Sorry, I'm just confused about what purpose this file serves

Comment (sammkj) [inline typings/glamorous.test.tsx]: Sorry I think I misunderstood your question. So this file is used to validate the type definitions by running `npm run test:ts`. This command will fail if type definitions are not correct.

Comment (beeman) [inline typings/glamorous.test.tsx]: It's referred in package.json afaics:
```
"test:ts": "tsc ./typings/glamorous.test.tsx --noEmit --jsx react --target es6 --module es2015 --moduleResolution node",
```

Comment (kentcdodds) [inline typings/glamorous.test.tsx]: Ah, perfect! Thanks!

**Comment by kentcdodds** [comment]
@sammkj, thanks so much for your help! I've added you as a collaborator on the project. Please make sure that you review the CONTRIBUTING.md file (specifically the bit about the commit messages and the git hooks) and familiarize yourself with the code of conduct (we're using [the contributor covenant](http://contributor-covenant.org/)). You might also want to [watch the repo](https://help.github.com/articles/watching-repositories/) to be notified when someone files an issue/PR. Please continue to make PRs as you feel the need (you can make your branches directly on the repo rather than your fork if you want). Thanks! And welcome to the team :)

File: typings/glamorous.test.tsx (hunk 1 above)
Comment (bterlson) [inline typings/glamorous.test.tsx]: Note that it doesn't validate whether the declaration file is accurate with respect to the implementation

Comment (kentcdodds) [inline typings/glamorous.test.tsx]: Thanks for the insights friends!

**Comment by kentcdodds** [comment]
@sammkj, I just realized that you didn't add yourself to the contributors table. Would you like to? See the contributing.md file 👍

**Comment by sammkj** [comment]
@kentcdodds Sure, I will add myself in the next PR.

**Pull request merged**